### PR TITLE
Update Cloud Reference for minor upgrades feature

### PR DESCRIPTION
**Pull request by matthijskrul**
## Summary of the changes / Why this is an improvement
- Updates the CrateDB Cloud Console overview to reflect the user-driven minor upgrades feature
- Updates the Glossary to explain versioning and link to CrateDB release notes

## Checklist

 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by matthijskrul** [comment]
NB: Please check whether I have defined the CrateDB versioning policy correctly

File: docs/glossary.rst [hunk 1]
```diff
@@ -478,10 +478,42 @@ defined role within the organization (see our reference on :ref:`user roles
 
     :ref:`User roles <user-roles>`
 
+.. _gloss-version:
+
+Versions
+--------
+
+CrateDB uses a versioning system with three levels of versioning: major
```
Comment (kojinkai) [inline docs/glossary.rst:486]: you could optionally make a reference to the [Semver](https://semver.org/) versioning system here that might already be familiar to some users.

So something like 

```suggestion
CrateDB uses a semantic versioning system, known as [Semver](https://semver.org/).  This system comprises  three levels of versioning: major
```

File: docs/glossary.rst [hunk 2]
```diff
@@ -478,10 +478,42 @@ defined role within the organization (see our reference on :ref:`user roles
...
+versions, minor versions, and patch versions. (Versions can also be referred to
+as releases.) These apply also to CrateDB Cloud clusters.
```
Comment (kojinkai) [inline docs/glossary.rst:488]: I am wondering if this line is a little ambiguous and could suggest to some users that we version the CrateDB Cloud software in the same way. Could this offer a bit more clarity?

```suggestion
as releases.) We support upgrading the CrateDB version that your cluster is running .

File: docs/glossary.rst [hunk 3]
```diff
@@ -478,10 +478,42 @@ defined role within the organization (see our reference on :ref:`user roles
...
+:ref:`Cluster Preferences <overview-cluster-settings-upgrade>`.
+
+A patch version of CrateDB is a release that includes bug fixes and smaller
+quality of life improvements compared to the previous such version. It is
+indicated by the third numeral in the versioning sequence, e.g. the 7 in
+'version 4.6.7'. You can upgrade patch versions yourself in the CrateDB Cloud
+Console under :ref:`Cluster Preferences <overview-cluster-settings-upgrade>`.
+
+.. SEEALSO::
+
```
Comment (kojinkai) [inline docs/glossary.rst:509]: I am wondering if we should explain that we only support  upgrading only one minor version at a time?

**Review by kojinkai — APPROVED** [review]
Looks good, just a couple of minor suggestions.

File: docs/glossary.rst (hunk 3 above)
Comment (matthijskrul) [reply docs/glossary.rst:509]: I do mention that in the Console Overview section it links to - it just seemed to me that it's not really relevant for the glossary. But you can argue it either way

File: docs/glossary.rst (hunk 2 above)
Comment (matthijskrul) [reply docs/glossary.rst:488]: Good point. I'll rephrase that

**Pull request merged**